server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CounterResource.java

```diff
@@ -0,0 +1,246 @@
+          .setDefaultValue(new ModelNode().set(0))
+          .build();
+
+    static final SimpleAttributeDefinition VALUE = new SimpleAttributeDefinitionBuilder(ModelKeys.VALUE,
+          ModelType.LONG, true)
+          .setXmlName(Attribute.VALUE.getLocalName())
+          .setAllowExpression(false)
+          .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+          .build();
+
+    static final AttributeDefinition[] ATTRIBUTES = { COUNTER_NAME, STORAGE, INITIAL_VALUE };
+
+    // operations
+
+    private static final OperationDefinition COUNTER_RESET = buildOperation("counter-reset").build();
+    private static final OperationDefinition COUNTER_INCREASE = buildOperation("counter-increase").build();
+    private static final OperationDefinition COUNTER_DECREASE = buildOperation("counter-decrease").build();
+
+    private final boolean runtimeRegistration;
+
+    public CounterResource(PathElement pathElement, ResourceDescriptionResolver descriptionResolver,
+            AbstractAddStepHandler addHandler, OperationStepHandler removeHandler, boolean runtimeRegistration) {
+        super(pathElement, descriptionResolver, addHandler, removeHandler);
+        this.runtimeRegistration = runtimeRegistration;
+    }
+
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        super.registerAttributes(resourceRegistration);
+        final OperationStepHandler writeHandler = new ReloadRequiredWriteAttributeHandler(ATTRIBUTES);
+        for (AttributeDefinition attr : ATTRIBUTES) {
+            resourceRegistration.registerReadWriteAttribute(attr, null, writeHandler);
+        }
+
+        if (runtimeRegistration) {
+            CounterMetricsHandler.INSTANCE.registerMetrics(resourceRegistration);
+        }
+    }
+
+    @Override
+    public void registerOperations(ManagementResourceRegistration resourceRegistration) {
+        super.registerOperations(resourceRegistration);
+        if (runtimeRegistration) {
+            resourceRegistration.registerOperationHandler(CounterResource.COUNTER_RESET, CounterResetCommand.INSTANCE);
+            resourceRegistration.registerOperationHandler(CounterResource.COUNTER_INCREASE,
+                    CounterIncreaseCommand.INSTANCE);
+            resourceRegistration.registerOperationHandler(CounterResource.COUNTER_DECREASE,
+                    CounterDecreaseCommand.INSTANCE);
+        }
+    }
+
+    private static SimpleOperationDefinitionBuilder buildOperation(String name) {
+        return new SimpleOperationDefinitionBuilder(name, new InfinispanResourceDescriptionResolver(ModelKeys.COUNTERS))
+                .setRuntimeOnly();
+    }
+
+    private static PathElement counterElement(ModelNode operation)
+            throws OperationFailedException {
+        final PathAddress address = pathAddress(operation.require(OP_ADDR));
+        final PathElement counterElement = address.getElement(address.size() - 1);
```

Comment:
you don't need the `counterElement`: `return address.getElement(address.size() - 1);`